### PR TITLE
Sort default components

### PR DESCRIPTION
**Pull request by remcohaszing**
This makes the output a bit more predictable.

### CODE REVIEW

**Comment by wooorm** [comment]
- Why only `tags`?
- a direct `sort`, because it mutates *and* returns, always seems a bit finicky to me, maybe a `const tags = [...scope.tags].sort()`?

**Comment by remcohaszing** [comment]
- What more than `tags`?
- It’s mutating a variable that only exists in the local scope, so I figured it’s ok. I can create a shallow copy if you still think that’s better.

**Comment by wooorm** [comment]
There’s also `objects`, `components`, and `idToInvalidComponentName` in [`scope`](https://github.com/mdx-js/mdx/blob/cb427e256848dd118ed47b26d14029e7ca138c0f/packages/mdx/lib/plugin/recma-jsx-rewrite.js#L35-L39). `references` is already sorted it looks like!

> I can create a shallow copy if you still think that’s better.

Yeah I think it‘s slightly better, but it’s mostly theoretical yeah

**Comment by remcohaszing** [comment]
I decided to sort the `actual` array, which is where `objects`, and `components` end up.

I also noticed some of these arrays contain unique values, so they would ideally be sets instead of arrays.

File: packages/mdx/lib/plugin/recma-jsx-rewrite.js [hunk 1]
```diff
@@ -363,18 +365,22 @@ export function recmaJsxRewrite(options) {
...
                 // `const _component0 = _components['custom-element']`
                 declarations.push({
                   type: 'VariableDeclarator',
-                  id: {type: 'Identifier', name: componentName},
+                  id: {
+                    type: 'Identifier',
+                    name: /** @type {string} */ (
+                      scope.idToInvalidComponentName.get(id)
+                    )
+                  },
```
Comment (wooorm) [inline packages/mdx/lib/plugin/recma-jsx-rewrite.js:383]: It’s a bit sad this cast is needed, as we could use `.entries` above, though then that would make sorting a bit harder.

File: packages/mdx/test/compile.js [hunk 2]
```diff
@@ -792,8 +792,8 @@ test('jsx', async () => {
       '/*@jsxRuntime automatic @jsxImportSource react*/',
       'function _createMdxContent(props) {',
       '  const _components = Object.assign({',
-      '    p: "p",',
-      '    em: "em"',
+      '    em: "em",',
+      '    p: "p"',
       '  }, props.components);',
```
Comment (wooorm) [inline packages/mdx/test/compile.js:797]: Is really only this one test case affected? I’d imagine a ton more would change?

File: packages/mdx/lib/plugin/recma-jsx-rewrite.js (hunk 1 above)
Comment (remcohaszing) [reply packages/mdx/lib/plugin/recma-jsx-rewrite.js:381]: Actually that’s not too bad. I decided to change it.

File: packages/mdx/test/compile.js (hunk 2 above)
Comment (remcohaszing) [reply packages/mdx/test/compile.js:797]: Yes. I was surprised too!

**Review by wooorm — APPROVED** [review]

**Pull request merged**